packages/core/package.json

```diff
@@ -15,7 +22,8 @@
     "lib/**/*.js",
     "lib/**/*.d.ts",
     "es/**/*.js",
-    "es/**/*.d.ts"
+    "es/**/*.d.ts",
+    "*.d.ts"
```

Comment:
Is this one needed?

Reply:
The plan was to keep types targeting lowe versions of TS in this file and that would have to be published. However, it seems that it's not possible to redirect internal file requests (see [here](https://github.com/microsoft/TypeScript-Website/pull/2344/files#r852949086)) so this strategy won't work.

To make it happen we would have to bundle all of our types, replace in it what we want, and point to that from `typesVersions`. It's way more work than what I've wanted to pull off here so I'm going to just revert this part of the change (optimization) and keep it in a TODO comment for v5.